projects/plugins/jetpack/extensions/blocks/premium-content/buttons/edit.js

```diff
@@ -59,14 +55,22 @@ function ButtonsEdit( { context, subscribeButton, setSubscribeButtonPlan } ) {
 	// Keep in sync the plan selected on the Premium Content block with the plan selected on the Recurring Payments
 	// inner block acting as a subscribe button.
 	useEffect( () => {
-		if ( ! planId || ! subscribeButton ) {
+		if ( ! planIds || ! subscribeButton ) {
 			return;
 		}
 
-		if ( subscribeButton.attributes.planId !== planId ) {
-			setSubscribeButtonPlan( planId );
+		const subscribeButtonPlanIds = subscribeButton.attributes.planIds || [];
+
+		// if the plan IDs are the same, don't update the subscribe button
+		if (
+			subscribeButtonPlanIds.length === planIds.length &&
+			subscribeButtonPlanIds.every( i => planIds.includes( i ) )
+		) {
+			return;
 		}
-	}, [ planId, subscribeButton, setSubscribeButtonPlan ] );
+
+		setSubscribeButtonPlan( planIds );
+	}, [ planIds, subscribeButton, setSubscribeButtonPlan ] );
```

Comment:
`setSubscribeButtonPlan` can be removed as it's not used inside the function.

Reply:
Yes it is (see line 72).